src/main/java/com/iexec/core/worker/WorkerService.java

```diff
@@ -50,15 +60,47 @@ public WorkerService(WorkerRepository workerRepository,
         this.contextualLockRunner = new ContextualLockRunner<>();
     }
 
+    @PostConstruct
+    void init() {
+        aliveWorkersGauge = Metrics.gauge(METRIC_WORKERS_GAUGE, new AtomicInteger(getAliveWorkers().size()));
+        aliveTotalCpuGauge = Metrics.gauge(METRIC_CPU_TOTAL_GAUGE, new AtomicInteger(getAliveTotalCpu()));
+        aliveAvailableCpuGauge = Metrics.gauge(METRIC_CPU_AVAILABLE_GAUGE, new AtomicInteger(getAliveAvailableCpu()));
```

Comment:
I think I would prefer to have an init at `0` here. Then, `true` values will be populated by the scheduled mechanism.
It would feel weird to me to see a scheduler starting with `10` workers to discover that only `4` are actually connected.

Reply:
After some more thinking, the proposed implementation is really accurate. It will fetch the correct number of workers after a scheduler restart. If the scheduler was offline for a long time, no alive workers will be seen.

Reply:
so nothing to do on my side ?

Reply:
nope, forget about it